src/coreclr/vm/methodtable.cpp

```diff
@@ -8012,28 +8022,33 @@ MethodTable::MethodData *MethodTable::GetMethodDataHelper(MethodTable *pMTDecl,
         }
     }
 
+    if (computeOptions == MethodDataComputeOptions::CacheOnly)
```

Comment:
This flags seems to be dependent on the state of the `s_fUseMethodDataCache`. Is that an intuitive assumption for this type?

Reply:
Yes...ish? It means look in the cache only for a result... and if `s_UseMethodDataCache` is false, that means the cache is always logically empty. Of course your question has inspired me to look deeper into `s_fUseMethodDataCache` and I've realized its ridiculous, and just removed it. Any .NET app of any reasonable size will have it set during the process eventually, and the current value of that api is to make it possible to hide some runtime bugs. I disapprove of that concept, so I'm going to enable the cache unconditionally.